lxd/main_init_interactive.go

```diff
@@ -0,0 +1,490 @@
+				if network.Config["bridge.external_interfaces"] != "" {
+					// Dummy validator for allowing empty strings
+					validator := func(string) error { return nil }
+					newNetwork.Config["bridge.external_interfaces"] = cli.AskString(
+						fmt.Sprintf(`Choose the local network interface to connect to network "%s" (empty for none): `, network.Name), "", validator)
+				}
+
+				config.Networks = append(config.Networks, newNetwork)
+			}
+		} else {
+			// New cluster
+			config.Cluster.ClusterPassword = cli.AskPassword("Trust password for new clients: ")
+		}
+	}
+
+	return nil
+}
+
+func (c *cmdInit) askNetworking(config *initData, d lxd.ContainerServer) error {
+	if !cli.AskBool("Would you like to create a new network bridge (yes/no) [default=yes]? ", "yes") {
+		return nil
+	}
+
+	for {
+		// Define the network
+		network := api.NetworksPost{}
+		network.Config = map[string]string{}
+
+		// Network name
+		network.Name = cli.AskString("What should the new bridge be called [default=lxdbr0]? ", "lxdbr0", networkValidName)
+		_, _, err := d.GetNetwork(network.Name)
+		if err == nil {
+			fmt.Printf("The requested network bridge \"%s\" already exists. Please choose another name.\n", network.Name)
+			continue
+		}
+
+		// Add to the default profile
+		config.Profiles[0].Devices["eth0"] = map[string]string{
+			"type":    "nic",
+			"nictype": "bridged",
+			"name":    "eth0",
+			"parent":  network.Name,
+		}
+
+		// IPv4
+		network.Config["ipv4.address"] = cli.AskString("What IPv4 address should be used (CIDR subnet notation, “auto” or “none”) [default=auto]? ", "auto", func(value string) error {
+			if shared.StringInSlice(value, []string{"auto", "none"}) {
+				return nil
+			}
+
+			return networkValidAddressCIDRV4(value)
+		})
+
+		if !shared.StringInSlice(network.Config["ipv4.address"], []string{"auto", "none"}) {
+			network.Config["ipv4.nat"] = fmt.Sprintf("%v",
+				cli.AskBool("Would you like LXD to NAT IPv4 traffic on your bridge? [default=yes]? ", "yes"))
+		}
+
+		// IPv6
+		network.Config["ipv6.address"] = cli.AskString("What IPv6 address should be used (CIDR subnet notation, “auto” or “none”) [default=auto]? ", "auto", func(value string) error {
+			if shared.StringInSlice(value, []string{"auto", "none"}) {
+				return nil
+			}
+
+			return networkValidAddressCIDRV6(value)
+		})
+
+		if !shared.StringInSlice(network.Config["ipv6.address"], []string{"auto", "none"}) {
+			network.Config["ipv6.nat"] = fmt.Sprintf("%v",
+				cli.AskBool("Would you like LXD to NAT IPv6 traffic on your bridge? [default=yes]? ", "yes"))
+		}
+
+		// Add the new network
+		config.Networks = append(config.Networks, network)
+		break
+	}
+
+	return nil
+}
+
+func (c *cmdInit) askStorage(config *initData, d lxd.ContainerServer) error {
+	if !cli.AskBool("Do you want to configure a new storage pool (yes/no) [default=yes]? ", "yes") {
+		return nil
+	}
+
+	// Figure out the preffered storage driver
```

Comment:
s/preffered/prefered/

Reply:
Yours is also wrong :) The right spelling is preferred

Reply:
fixed